tests/testthat/test-derive_date_vars.R

```diff
@@ -0,0 +1,895 @@
+library(lubridate)
+library(dplyr)
+
+input <- c(
+  "2019-07-18T15:25:40.243",
+  "2019-07-18T15:25:40",
+  "2019-07-18T15:25",
+  "2019-07-18",
+  "2019-02",
+  "2019",
+  "2019---07",
+  "2003-12-15T-:15:18",
+  "2003-12-15T13:-:19",
+  "2020-07--T00:00"
+)
+
+# impute_dtc_dtm ----
+## Test 1: default: no date imputation, time part set to 00:00:00 ----
+test_that("impute_dtc_dtm Test 1: default: no date imputation, time part set to 00:00:00", {
+  expected_output <- c(
+    "2019-07-18T15:25:40.243",
+    "2019-07-18T15:25:40",
+    "2019-07-18T15:25:00",
+    "2019-07-18T00:00:00",
+    NA_character_,
+    NA_character_,
+    NA_character_,
+    "2003-12-15T00:00:00",
+    "2003-12-15T13:00:00",
+    NA_character_
+  )
+  expect_equal(impute_dtc_dtm(dtc = input), expected_output)
+})
+
+## Test 2: no date imputation, min and sec imputed with 59 ----
+test_that("impute_dtc_dtm Test 2: no date imputation, min and sec imputed with 59", {
+  expected_output <- c(
+    "2019-07-18T15:25:40.243",
+    "2019-07-18T15:25:40",
+    "2019-07-18T15:25:59",
+    NA_character_,
+    NA_character_,
+    NA_character_,
+    NA_character_,
+    NA_character_,
+    "2003-12-15T13:59:59",
+    NA_character_
+  )
+  expect_equal(
+    impute_dtc_dtm(
+      dtc = input,
+      highest_imputation = "m",
+      time_imputation = "23:59:59"
+    ),
+    expected_output
+  )
+
+  expect_equal(
+    impute_dtc_dtm(
+      dtc = input,
+      highest_imputation = "m",
+      time_imputation = "LAST"
+    ),
+    expected_output
+  )
+})
+
+## Test 3: impute date to first, time to 00:00:00 ----
+test_that("impute_dtc_dtm Test 3: impute date to first, time to 00:00:00", {
+  expected_output <- c(
+    "2019-07-18T15:25:40.243",
+    "2019-07-18T15:25:40",
+    "2019-07-18T15:25:00",
+    "2019-07-18T00:00:00",
+    "2019-02-01T00:00:00",
+    "2019-01-01T00:00:00",
+    "2019-01-01T00:00:00",
+    "2003-12-15T00:00:00",
+    "2003-12-15T13:00:00",
+    "2020-07-01T00:00:00"
+  )
+
+  expect_equal(
+    impute_dtc_dtm(
+      dtc = input,
+      highest_imputation = "M",
+      date_imputation = "first"
+    ),
+    expected_output
+  )
+
+  expect_equal(
+    impute_dtc_dtm(
+      dtc = input,
+      highest_imputation = "M",
+      date_imputation = "01-01"
+    ),
+    expected_output
+  )
+})
+
+## Test 4: impute day to last, time to 23:59:59 ----
+test_that("impute_dtc_dtm Test 4: impute day to last, time to 23:59:59", {
+  expected_output <- c(
+    "2019-07-18T15:25:40.243",
+    "2019-07-18T15:25:40",
+    "2019-07-18T15:25:59",
+    "2019-07-18T23:59:59",
+    "2019-02-28T23:59:59",
+    NA_character_,
+    NA_character_,
+    "2003-12-15T23:59:59",
+    "2003-12-15T13:59:59",
+    "2020-07-31T23:59:59"
+  )
+  expect_equal(
+    impute_dtc_dtm(
+      dtc = input,
+      highest_imputation = "D",
+      date_imputation = "LAST",
```

Comment:
minor comment as i know these are only unit tests but lets use the same casing across all our example calls within our repo, i.e. "last"

Reply:
Updated